[ISSUE] Issue with `databricks_group_member` resource

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Databricks Provider issue tracker reserved for bug reports and feature requests. For general usage questions, please see: <https://www.terraform.io/community.html>.
-->
I have problems adding a service-principal to an account-level group. To eliminate errors I hard-coded the IDs, but even then it dont work.
### Configuration
<!-- Please provide a minimal reproducible configuration for the issue -->
```hcl
resource "databricks_group_member" "cicd_sp_in_databricks_group" {
  group_id  = "778143257983011" # This is the group ID from accounts.azuredatabricks.net - groups - THE_GROUP - Group Information
  member_id = "67f5fc18-1a8c-4c4b-b7b9-773db16bcceb" # This is the UUID from  accounts.azuredatabricks.net - Service principals - THE_SP - Principal information - UUID
}

```

### Expected Behavior
The SP should be added to the account level group

### Actual Behavior
I get the responce

> ╷
│ Error: cannot read group member: Group has no member
│
│   with databricks_group_member.cicd_sp_in_databricks_group,
│   on prelect_groups.tf line 17, in resource "databricks_group_member" "cicd_sp_in_databricks_group":
│   17: resource "databricks_group_member" "cicd_sp_in_databricks_group" {
│
╵

### Steps to Reproduce
<!-- Please list the steps required to reproduce the issue, for example:
1. `terraform apply`-->

### Terraform and provider versions
<!-- Please paste the output of `terraform version`. If version of `databricks` provider is not the latest (https://github.com/databricks/terraform-provider-databricks/releases), please make sure to use the latest one. -->

Terraform v1.9.5
on linux_amd64
+ provider registry.terraform.io/databricks/databricks v1.56.0
+ provider registry.terraform.io/hashicorp/azurerm v3.97.1                                                                                                                                                                                                                              

### Is it a regression?
<!-- Did this work in a previous version of the provider? If so, which versions did you try? -->
Not tested before.

### Debug Output
<!-- Please add turn on logging, e.g. `TF_LOG=DEBUG terraform apply -no-color` and run command again, paste it to gist & provide the link to gist. If you're still willing to paste in log output, make sure you provide only relevant log lines with requests. It would make it more readable, if you pipe the log through `| grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'`, e.g.: `TF_LOG=DEBUG terraform apply -no-color 2>&1 | grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/' 2>&1 |tee tf-debug.log`. If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. -->
The log entries right before the error is as follows (name changed). It prints out all the existing members of the group

< ...
<       "$ref": "Users/7961626071215147",
<       "display": "FAKE NAMESEN",
<       "value": "7961626071215147"
<     },
<     "... (5 additional elements)"
<   ],
<   "schemas": [
<     "urn:ietf:params:scim:schemas:core:2.0:Group"
<   ]
< }: @module=databricks tf_provider_addr=registry.terraform.io/databricks/databricks tf_rpc=ConfigureProvider tf_mux_provider=tf5to6server.v5tov6Server tf_req_id=da00e29a-0e10-9556-8da4-86d616528c06 @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/logger/logger.go:38 timestamp="2024-11-05T16:51:40.305+0100"
2024-11-05T16:51:40.307+0100 [ERROR] provider.terraform-provider-databricks_v1.56.0: Response contains error diagnostic: tf_req_id=bd02beb9-0f6e-1be7-60f4-23e86a2a256d tf_rpc=ApplyResourceChange @caller=/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_detail="" diagnostic_severity=ERROR diagnostic_summary="cannot read group member: Group has no member" tf_provider_addr=registry.terraform.io/databricks/databricks @module=sdk.proto tf_proto_version=6.6 tf_resource_type=databricks_group_member timestamp="2024-11-05T16:51:40.306+0100"
2024-11-05T16:51:40.309+0100 [DEBUG] State storage *remote.State declined to persist a state snapshot
2024-11-05T16:51:40.309+0100 [ERROR] vertex "databricks_group_member.cicd_sp_in_databricks_group" error: cannot read group member: Group has no member
2024-11-05T16:51:40.310+0100 [DEBUG] states/remote: state read serial is: 167; serial is: 167
2024-11-05T16:51:40.310+0100 [DEBUG] states/remote: state read lineage is: eb3984a7-a6fe-5105-5e7c-3de2355a0f0c; lineage is: eb3984a7-a6fe-5105-5e7c-3de2355a0f0c
╷

### Important Factoids
<!-- Are there anything atypical about your accounts that we should know? -->
The group is an account level group, and the terraform is run by a CICD service-principal with account-level access. 
The host is `  host       = "https://accounts.azuredatabricks.net/"`

--- a/NEXT_CHANGELOG.md
+++ b/NEXT_CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### Documentation
 
+ * Explicitly mention SCIM ID in `databricks_group_member` docs [#4709](https://github.com/databricks/terraform-provider-databricks/pull/4709)
+
 ### Exporter
 
 ### Internal Changes

--- a/docs/resources/group.md
+++ b/docs/resources/group.md
@@ -101,12 +101,12 @@ The following arguments are supported:
 
 In addition to all arguments above, the following attributes are exported:
 
-* `id` -  The id for the group object.
+* `id` - Canonical unique identifier for the group (SCIM ID).
 * `acl_principal_id` - identifier for use in [databricks_access_control_rule_set](access_control_rule_set.md), e.g. `groups/Some Group`.
 
 ## Import
 
-You can import a `databricks_group` resource with the name `my_group` like the following:
+You can import a `databricks_group` resource by its SCIM ID:
 
 ```hcl
 import {

--- a/docs/resources/group_member.md
+++ b/docs/resources/group_member.md
@@ -41,8 +41,8 @@ resource "databricks_group_member" "bb" {
 
 The following arguments are supported:
 
-* `group_id` - (Required) This is the id of the [group](group.md) resource.
-* `member_id` - (Required) This is the id of the [group](group.md), [service principal](service_principal.md), or [user](user.md).
+* `group_id` - (Required) This is the `id` attribute (SCIM ID) of the [group](group.md) resource.
+* `member_id` - (Required) This is the `id` attribute (SCIM ID) of the [group](group.md), [service principal](service_principal.md), or [user](user.md).
 
 ## Attribute Reference
 

--- a/docs/resources/service_principal.md
+++ b/docs/resources/service_principal.md
@@ -114,14 +114,14 @@ The following arguments are available:
 
 In addition to all arguments above, the following attributes are exported:
 
-- `id` - Canonical unique identifier for the service principal.
+- `id` - Canonical unique identifier for the service principal (SCIM ID).
 - `home` - Home folder of the service principal, e.g. `/Users/00000000-0000-0000-0000-000000000000`.
 - `repos` - Personal Repos location of the service principal, e.g. `/Repos/00000000-0000-0000-0000-000000000000`.
 - `acl_principal_id` - identifier for use in [databricks_access_control_rule_set](access_control_rule_set.md), e.g. `servicePrincipals/00000000-0000-0000-0000-000000000000`.
 
 ## Import
 
-The resource scim service principal can be imported using its id, for example `2345678901234567`. To get the service principal ID, call [Get service principals](https://docs.databricks.com/dev-tools/api/latest/scim/scim-sp.html#get-service-principals).
+The resource scim service principal can be imported using its SCIM id, for example `2345678901234567`. To get the service principal ID, call [Get service principals](https://docs.databricks.com/dev-tools/api/latest/scim/scim-sp.html#get-service-principals).
 
 ```hcl
 import {

--- a/docs/resources/user.md
+++ b/docs/resources/user.md
@@ -109,14 +109,14 @@ The following arguments are available:
 
 In addition to all arguments above, the following attributes are exported:
 
-* `id` - Canonical unique identifier for the user.
+* `id` - Canonical unique identifier for the user (SCIM ID).
 * `home` - Home folder of the user, e.g. `/Users/mr.foo@example.com`.
 * `repos` - Personal Repos location of the user, e.g. `/Repos/mr.foo@example.com`.
 * `acl_principal_id` - identifier for use in [databricks_access_control_rule_set](access_control_rule_set.md), e.g. `users/mr.foo@example.com`.
 
 ## Import
 
-The resource scim user can be imported using id:
+The resource scim user can be imported using its SCIM id:
 
 ```hcl
 import {

